src/Dfe.PlanTech.AzureFunctions/Functions/QueueReceiver.cs

```diff
@@ -11,162 +11,198 @@
+        foreach (ServiceBusReceivedMessage message in messages)
         {
-            Logger.LogInformation("Queue Receiver -> Db Writer started. Processing {msgCount} messages", messages.Length);
-
-            foreach (ServiceBusReceivedMessage message in messages)
-            {
-                await ProcessMessage(message, messageActions, cancellationToken);
-            }
+            await ProcessMessage(message, messageActions, cancellationToken);
         }
+    }
 
-        private async Task ProcessMessage(ServiceBusReceivedMessage message, ServiceBusMessageActions messageActions, CancellationToken cancellationToken)
+    private async Task ProcessMessage(ServiceBusReceivedMessage message, ServiceBusMessageActions messageActions, CancellationToken cancellationToken)
+    {
+        try
         {
-            try
-            {
-                string cmsEvent = GetCmsEvent(message.Subject);
-                var text = Encoding.UTF8.GetString(message.Body);
+            string cmsEvent = GetCmsEvent(message.Subject);
 
-                Logger.LogInformation("Performing Action: {action}", cmsEvent);
-                Logger.LogInformation("Processing {text}", text);
+            Logger.LogInformation("CMS Event: {cmsEvent}", cmsEvent);
 
-                ContentComponentDbEntity mapped = _mappers.ToEntity(text);
-                ContentComponentDbEntity? existing = await GetExistingDbEntity(mapped, cancellationToken);
-
-                if (existing != null)
-                {
-                    mapped.Archived = existing.Archived;
-                    mapped.Published = existing.Published;
-                    mapped.Deleted = existing.Deleted;
-                }
-
-                switch (cmsEvent)
-                {
-                    case "create":
-                    case "save":
-                    case "auto_save":
-                        break;
-                    case "archive":
-                        mapped.Archived = true;
-                        break;
-                    case "unarchive":
-                        mapped.Archived = false;
-                        break;
-                    case "publish":
-                        mapped.Published = true;
-                        break;
-                    case "unpublish":
-                        mapped.Published = false;
-                        break;
-                    case "delete":
-                        mapped.Deleted = true;
-                        break;
-                    default:
-                        throw new CmsEventException(string.Format("CMS Event \"{0}\" not implemented", cmsEvent));
-                }
+            ContentComponentDbEntity mapped = GetMapped(message);
+            ContentComponentDbEntity? existing = await GetExisting(mapped, cancellationToken);
 
-                long rowsChanged = await UpsertEntityInDatabase(mapped, existing, cancellationToken);
+            if (cmsEvent.Equals("unpublish") && existing != null)
```

Comment:
Can we rejig this to combine it with the `ProcessCmsEvent` method? IMO it would be cleaner, otherwise technically you process the event here, _and_ in the `ProcessCmsEvent`